Lib/urllib2.py

```diff
@@ -858,7 +858,7 @@ class AbstractBasicAuthHandler:
 
     # allow for double- and single-quoted realm values
     # (single quotes are a violation of the RFC, but appear in the wild)
-    rx = re.compile('(?:.*,)*[ \t]*([^ \t]+)[ \t]+'
+    rx = re.compile('(?:.*,)*[ \t]*([^ \t,]+)[ \t]+'
```

Comment:
We could also fix CVE-2020-8492.

Replacing `(?:.*,)*` with `(?:[^,]*,)*` doesn't change the parser and it makes the parser more efficient. See the first message of the issue which proposes this change: https://bugs.python.org/issue39503#msg361072

```suggestion
    rx = re.compile('(?:[^,]*,)*[ \t]*([^ \t,]+)[ \t]+'
```

Example:

```
>>> import re
>>> tests=['a,b,', 'a,' 'abc,def,ghi,']
# Old regex
>>> all(re.match(r'(?:.*,)*', item) is not None for item in tests)
True
# New regex
>>> all(re.match(r'(?:[^,]*,)*', item) is not None for item in tests)
True
```

Reply:
With that fix, the performance is much better. Are you sure that this change does not break the original "buggy" behavior we have to preserve? We can discuss it on triage and reopen bugs for 2020-8492 if we decide to fix them both.

Reply:
The two regular expressions match the same strings. The only difference is the performance.